dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/common/hibernate/HibernateIdentifiableObjectStore.java

```diff
@@ -188,16 +199,33 @@ public void update(@Nonnull T object, @CheckForNull User user) {
     getSession().update(object);
   }
 
-  @Override
-  public void delete(@Nonnull T object) {
-    this.delete(object, currentUserService.getCurrentUser());
+  private void setFields(T object, UserDetails userDetails) {
+    checkNotNull(object);
+    checkNotNull(userDetails);
+
+    object.setAutoFields();
+
+    // TODO: MAS: id=0 should not be necessary, only needed for tests
+    // TODO: MAS: Replace all usage of id=0 with use of a SystemUser instance
+    // issues with transaction isolation
+    if (userDetails.getId() != 0L && !(userDetails instanceof SystemUser)) {
+
+      // See: https://www.baeldung.com/jpa-entity-manager-get-reference
+      User user = entityManager.getReference(User.class, userDetails.getId());
+      object.setLastUpdatedBy(user);
```

Comment:
does this mean that `lastUpdatedBy` will never show as system user? just wondering for my own understanding.

Reply:
Yes, this is correct.